packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybrid2GL.hpp

```diff
@@ -0,0 +1,1532 @@
+	comm_time += curr_comm_time;
+
+        if(distributedRounds < numStatisticRecordingRounds){
+          commPerRound[distributedRounds] = curr_comm_time;
+	  recvPerRound[distributedRounds] = recv;
+          sentPerRound[distributedRounds] = sent;
+	  if(verbose) {
+            std::cout<<comm->getRank()<<": total sent in round "<<distributedRounds<<" = "<<sent<<"\n";
+            std::cout<<comm->getRank()<<": total recv in round "<<distributedRounds<<" = "<<recv<<"\n";
+	  }
+          totalPerRound[distributedRounds] += commPerRound[distributedRounds];
+	}
+
+        Kokkos::parallel_for(n_ghosts, KOKKOS_LAMBDA(const int& i){
+          ghost_colors(i) = femv_colors(i+n_local);
+        });
+        Kokkos::fence();
+        comm->barrier();
+
+        //these views will change in detectConflicts, they need
+	//to be zeroed out beforehand
+        verts_to_send_size_host(0) = 0;
+	verts_to_recolor_size_host(0) = 0;
+	recoloringSize_host(0) = 0;
+	Kokkos::deep_copy(verts_to_send_size, verts_to_send_size_host);
+	Kokkos::deep_copy(verts_to_recolor_size, verts_to_recolor_size_host);
+	Kokkos::deep_copy(recoloringSize, recoloringSize_host);
+
+        //check for further conflicts
+        double detection_temp = timer();
+
+        detectConflicts(n_local, dist_offsets_dev, dist_adjs_dev,femv_colors, boundary_verts_dev, boundary_size,
+	         verts_to_recolor_atomic, verts_to_recolor_size_atomic, verts_to_send_atomic, verts_to_send_size_atomic, 
+		 recoloringSize, rand_dev, gid_dev, ghost_degrees_dev, recolor_degrees);
+
+        //copy the updated device views back into host memory where necessary
+        Kokkos::deep_copy(verts_to_send_host, verts_to_send_view);
+        Kokkos::deep_copy(verts_to_send_size_host, verts_to_send_size);
+	//we only use the list of verts to recolor on device, no need to copy to host.
+        Kokkos::deep_copy(verts_to_recolor_size_host, verts_to_recolor_size);
+	Kokkos::deep_copy(recoloringSize_host, recoloringSize);
+        
+	if(distributedRounds < numStatisticRecordingRounds){
+	  conflictDetectionPerRound[distributedRounds] = timer() - detection_temp;
+          conflict_detection += conflictDetectionPerRound[distributedRounds];
+          compPerRound[distributedRounds] += conflictDetectionPerRound[distributedRounds];
+          totalPerRound[distributedRounds] += conflictDetectionPerRound[distributedRounds];
+          comp_time += conflictDetectionPerRound[distributedRounds];
+        }
+
+        distributedRounds++;
+        int localDone = recoloringSize_host(0);
+        int globalDone = 0;
+        Teuchos::reduceAll<int,int>(*comm, Teuchos::REDUCE_SUM, 1, &localDone, &globalDone);
+        done = !globalDone;
+        
+      }//end device coloring
+      
+      if(recoloringSize_host(0) > 0 || !done){
+
+	ghost_colors_host = Kokkos::create_mirror_view(ghost_colors);
+	deep_copy(ghost_colors_host, ghost_colors);
```

Comment:
prefer create_mirror_view_and_copy to avoid initialization of ghost_colors_host before deep copy into it.

Reply:
Done.